README.Rmd

```diff
@@ -46,10 +46,84 @@ Run with profile from [memprof](https://github.com/mrc-ide/memprof)
 
 ```{r}
 source("threemc_fit.R")
-mem <- memprof::with_monitor(threemc_fit())
+# load shell dataset
+shell_dat <- readr::read_csv("data/shell_data_lso.csv.gz")
+# load shapefiles
+areas <- sf::read_sf("data/areas.geojson")
+# load text from C++ mod into R
+mod <- readLines("src/threemc.cpp")
+mem <- memprof::with_monitor(threemc_fit(shell_dat, areas, mod))
 plot(mem)
 ```
-## Full model fit
 
-Running the model with a larger country leads to huge spikes in memory usage > 500GB.
+## Cluster Example 
+
+The above example is for a small country, that can be run locally. However, 
+when we look at modelling a large country, such as Ghana, we must run the model 
+on a high performance cluster. The spike in memory shown above can even 
+exceed 500GB. 
+```{r}
+# load shell dataset for Ghana
+shell_dat_gha <- readr::read_csv("data/shell_data_gha.csv.gz")
+```
+
+```{r, results = "false"}
+# specify directory that 
+root <- "~/net/unaids-naomi/threemc-orderly/contexts_4"
+
+# cluster config
+config <- didehpc::didehpc_config(
+  workdir  = root,
+  cluster  = "fi--didemrchnb", 
+  # need to request every core of largest cluster nodes in order to fit model
+  # => at lower memory usage we could be running much more concurrent models!
+  template = "32Core",        
+  cores    = 32
+)
+
+# create symlink in cluster contexts dir to sourced functions
+source_file <- file.path(root, "threemc_fit.R")
+if (.Platform$OS.type != "windows" && !file.exists(source_file)) {
+  system(paste0("ln -s threemc_fit.R ", source_file))
+}
+
+# setup context for orderly task 
+ctx <- context::context_save(
+  path = root,
+  # functions to source in current directory
+  sources = "./threemc_fit.R",
+  # CRAN packages
+  packages = c("dplyr", "sf", "readr", "TMB", "stats"),
+  # packages from github 
+  package_sources = conan::conan_sources(c(
+    "github::mrc-ide/threemc", 
+    "github::mrc-ide/memprof"
+  ))
+)
+# queue above contexts
+obj <- didehpc::queue_didehpc(context = ctx, config = config)
+```
```

Comment:
I think I'd be tempted to leave this out of the final md as TMB guys won't be able to recreate it. Could keep it in the Rmd and put `include = FALSE` on the block

Reply:
Yeah agreed that it isn't re-creatable, just thought it might be useful to show how we have to command every core on each cluster node just to fit the model!

Reply:
Thanks, have put a note into the preamble at the top about this to explain our motivation. Have moved the cluster example into a separate Rmd and put the plot at the top